### PR TITLE
fix bugs

### PR DESCRIPTION
**Pull request by bdarcus**
This branch needs to fix the little bugs.

I started with required "form" attributes where they shouldn't be required, but have not finished.

There are a number of other errors one gets when running against any style; in my case, a chicago note one.

cc @bwiernik @denismaier

### CODE REVIEW

**Comment by bwiernik** [comment]
Thanks for getting this started. Once it is ready for merging, we will also want to merge into the 1.0 branch and update the 1.0.2 tag.

**Comment by bdarcus** [comment]
> Thanks for getting this started. Once it is ready for merging, we will also want to merge into the 1.0 branch and update the 1.0.2 tag.

Should we just delete that tag for now?

**Comment by bwiernik** [comment]
Good idea

File: schemas/styles/csl.rnc [hunk 1]
```diff
@@ -647,10 +647,11 @@ div {
       ## Specify verbatim text.
       attribute value { text }
     | ((attribute variable { variables.numbers | variables.strings },
-        [ a:defaultValue = "long" ] attribute form { "short" | "long" })
+        [ a:defaultValue = "long" ]
+        attribute form { "short" | "long" }?)
        | (attribute variable { variables.titles },
           [ a:defaultValue = "long" ]
-          attribute form { "short" | "long" | "sub" | "main" })?)
+          attribute form { "short" | "long" | "sub" | "main" }?)?)
```
Comment (denismaier) [inline schemas/styles/csl.rnc:654]: I'm a bit confused by this one. I thought "sub" and "main" on `@form` will be added in 1.1, not already in 1.0.2. Maybe we got confused with the different branches?

Comment (bwiernik) [reply schemas/styles/csl.rnc:654]: Yes, these were removed from 1.0.2

Comment (bwiernik) [reply schemas/styles/csl.rnc:654]: I don't recall exactly what the status of master vs 1.0 branches were once we finished. @bdarcus how does 1.0 run in your tests?

Comment (bdarcus) [reply schemas/styles/csl.rnc:654]: Oh, I was using master here. I guess I should have clarified that.

Was that wrong?

Comment (bwiernik) [reply schemas/styles/csl.rnc:654]: I think 1.0 (and the tag) was the official "1.0.2"

Comment (bwiernik) [reply schemas/styles/csl.rnc:654]: I'm recalling our frustration managing the various branches

Comment (denismaier) [reply schemas/styles/csl.rnc:654]: > Oh, I was using master here. I guess I should have clarified that.
> 
> Was that wrong?

Well, sort of. There's nothing wrong with fixing bugs on that branch, but at least it comes as no suprise that validating 1.0.1/1.0.2 styles fails with that branch:

grafik

Comment (denismaier) [reply schemas/styles/csl.rnc:654]: Hmm, I've got version 1.1 even with v1.0.2 checked out. It looks like that happened somewhere between v1.0.2-pre.2 and v1.0.2.  (Current styles seem to be valid with v1.0.2-pre.2, but Atom is throwing millions of errors with v1.0.2)

Comment (bdarcus) [reply schemas/styles/csl.rnc:654]: Hmm ...can I ask you two to figure out the best way forward here? It sounds like maybe we should close this PR and branch and start over?

I had a sudden death in the family this weekend, so I'm a bit distracted.

Comment (denismaier) [reply schemas/styles/csl.rnc:654]: I'll try to figure out what was happening between v1.0.2-pre.2 and v1.0.2.

Comment (denismaier) [reply schemas/styles/csl.rnc:654]: I'll open a new PR. Let's close this one for now.

**Pull request closed without merging**